source/base/NexusApp.cc

```diff
@@ -206,7 +206,7 @@ void NexusApp::ExecuteMacroFile(const char* filename)
   std::unique_ptr<G4UIsession> batchSession =
     std::make_unique<BatchSession>(filename, UI->GetSession());
   UI->SetSession(batchSession.get());
-  G4UIsession* previousSession = UI->GetSession()->SessionStart();
+  G4UIsession* previousSession = batchSession->SessionStart();
   UI->SetSession(previousSession);
```

Comment:
1. I recommend `using std::unique_ptr;` and `using std::make_unique;` at the top of this file, so that you can avoid having to qualify them with `std` all over the place. NB: do **NOT** do `using std::move`. Always fully qualify `std::move`.

2. Is there any good reason for writing
   ```c++
   std::unique_ptr<G4UIsession> batchSession = std::make_unique<BatchSession>(filename, UI->GetSession());
   ```
   instead of 
   ```c++
   auto batchSession = make_unique<BatchSession>(filename, UI->GetSession());
   ```
   ?

Reply:
> * Is there any good reason for writing
>   ```c++
>   std::unique_ptr<G4UIsession> batchSession = std::make_unique<BatchSession>(filename, UI->GetSession());
>   ```

>   instead of
>   ```c++
>   auto batchSession = make_unique<BatchSession>(filename, UI->GetSession());
>   ```     
>   ?

`batchSession` is passed as an argument of the `SetSession()` function of the `G4UIsession` class and this function needs a `G4UIsession` object, so I think that the reason was to make clear that this object is of type `G4UIsession`. Maybe it's not necessary.

Reply:
The type of `batchSession` is definitely `BatchSession` (give or take a pointer). `BatchSession` implements (is a subtype of) `G4UIsession` which is why it can be passed to functions which formally accept the latter. All this requires *very* basic understanding of C++ inheritance and absolutely anyone getting anywhere near C++ should be thoroughly aware of it, so I don't think we're doing anyone any favours by adding the extra noise.

If it were up to me, I'd get rid of it.